demo.py

```diff
@@ -25,115 +27,214 @@
+
+    G = nx.grid_2d_graph(w, h)
+    nodes = G.nodes()
+
+    # Identify a fixed set of points of interest
+    pois = random.sample(nodes, k=num_poi)
+
+    # Identify a fixed set of current charging locations
+    charging_stations = random.sample(nodes, k=num_cs)
+
+    # Identify potential new charging locations
+    potential_new_cs_nodes = list(nodes - charging_stations)
+
+    return G, pois, charging_stations, potential_new_cs_nodes
+
+def build_bqm(potential_new_cs_nodes, num_poi, pois, num_cs, charging_stations, num_new_cs):
+    """ Build bqm that models our problem scenario for the hybrid sampler. """
+
+    # Tunable parameters
+    gamma1 = len(potential_new_cs_nodes) * 4
+    gamma2 = len(potential_new_cs_nodes) / 3
+    gamma3 = len(potential_new_cs_nodes) * 1.7
+    gamma4 = len(potential_new_cs_nodes) ** 3
+
+    # Build BQM using adjVectors to find best new charging location s.t. min 
+    # distance to POIs and max distance to existing charging locations
+    bqm = dimod.AdjVectorBQM(len(potential_new_cs_nodes), 'BINARY')
+
+    # Constraint 1: Min average distance to POIs
+    if num_poi > 0:
+        for i in range(len(potential_new_cs_nodes)):
+            # Compute average distance to POIs from this node
+            avg_dist = 0
+            cand_loc = potential_new_cs_nodes[i]
+            for loc in pois:
+                dist = (cand_loc[0]**2 - 2*cand_loc[0]*loc[0] + loc[0]**2 
+                                    + cand_loc[1]**2 - 2*cand_loc[1]*loc[1] + loc[1]**2)
+                avg_dist += dist / num_poi 
+            bqm.linear[i] += avg_dist * gamma1
+
+    # Constraint 2: Max distance to existing chargers
+    if num_cs > 0:
+        for i in range(len(potential_new_cs_nodes)):
+            # Compute average distance to POIs from this node
+            avg_dist = 0
+            cand_loc = potential_new_cs_nodes[i]
+            for loc in charging_stations:
+                dist = (-1*cand_loc[0]**2 + 2*cand_loc[0]*loc[0] - loc[0]**2
+                                    - cand_loc[1]**2 + 2*cand_loc[1]*loc[1] - loc[1]**2)
+                avg_dist += dist / num_cs
+            bqm.linear[i] += avg_dist * gamma2
+
+    # Constraint 3: Max distance to other new charging locations
+    if num_new_cs > 1:
+        for i in range(len(potential_new_cs_nodes)):
+            for j in range(i+1, len(potential_new_cs_nodes)):
+                ai = potential_new_cs_nodes[i]
+                aj = potential_new_cs_nodes[j]
+                dist = (-1*ai[0]**2 + 2*ai[0]*aj[0] - aj[0]**2 - ai[1]**2 
+                        + 2*ai[1]*aj[1] - aj[1]**2)
+                bqm.add_interaction(i, j, dist * gamma3)
+
+    # Constraint 4: Choose exactly num_new_cs new charging locations
+    bqm.update(dimod.generators.combinations(bqm.variables, num_new_cs, strength=gamma4))
+
+    return bqm
+
+def run_bqm_and_collect_solns(bqm, sampler, potential_new_cs_nodes, **kwargs):
```

Comment:
It would be somewhat considered "best practices" and also consistent with other methods in the file to rename this to `run_bqm_and_collect_solutions`.